pythonFiles/runJediLanguageServer.py

```diff
@@ -5,8 +5,86 @@
 EXTENSION_ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 sys.path.append(os.path.join(EXTENSION_ROOT, "pythonFiles", "lib", "python"))
 
+import pygls.protocol
+
+
+def is_json_basic_type(obj):
+    """Returns true if the object is an instance of type [int, floor, bool, str]"""
+    return isinstance(obj, (int, float, bool, str))
+
+
+def remove_null_fields(obj, obj_field_name=None):
+    """This function is used to remove fields with a 'None' value. This is a temporary
+    workaround to address the following issue:
+
+    https://github.com/pappasam/jedi-language-server/issues/60
+    https://github.com/openlawlibrary/pygls/issues/145
+    https://github.com/microsoft/vscode-languageserver-node/issues/740
+
+    Essentially the LS Client in VS Code expects optional fields that are not needed
+    should be skipped. Currently in pygls options fields use 'null' value.
+    https://github.com/microsoft/vscode-languageserver-node/issues/740#issuecomment-773967897
+    """
+    if is_json_basic_type(obj):
+        return
+    elif isinstance(obj, list):
+        for o in obj:
+            remove_null_fields(o, obj_field_name)
+        return
+
+    # We need this as a list for now so we can check for the special case with
+    # textDocument object
+    attributes = list(
```

Comment:
This would be clearer and more concise with a list comprehension.

Reply:
it was a list comprehension originally. I will add it back.